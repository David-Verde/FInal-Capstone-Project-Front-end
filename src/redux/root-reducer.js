import categoryReducer from './category/category.reducer';
import propertyReducer from './property/property.reducer';
import reservationReducer from './reservation/reservation.reducer';
import statusReducer from './status/status.reducer';
import userReducer from './user/user.reducer';

export const rootReducer = {
  user: userReducer,
  property: propertyReducer,
  category: categoryReducer,
  reservation: reservationReducer,
  status: statusReducer,
};
