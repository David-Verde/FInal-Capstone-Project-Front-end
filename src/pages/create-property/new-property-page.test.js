import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '../../redux/store';
import NewPropertyPage from './newProperty-page.component';

const renderWithProviders = (reduxStore, route) => render(
  <Provider store={reduxStore}>
    <MemoryRouter initialEntries={[route]}>
      <NewPropertyPage />
    </MemoryRouter>
  </Provider>,
);

describe('NewPropertyPage component', () => {
  it('should render correctly', () => {
    const component = renderWithProviders(store, '/');
    expect(component).toMatchSnapshot();
  });
});
