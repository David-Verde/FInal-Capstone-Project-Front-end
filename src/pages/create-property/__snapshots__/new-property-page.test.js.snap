// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPropertyPage component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="new-property"
      >
        <section
          class="add-property"
        >
          <div
            class="add-property__header"
          >
            <h2>
              Add your property so that other people can rent it!
            </h2>
            <p>
              Please, fill all the inputs so that you can host a home!
            </p>
          </div>
          <form
            class="form"
          >
            <div
              class="form__text-containers"
            >
              <div
                class="form__inputBox"
              >
                <input
                  name="name"
                  placeholder=" "
                  required=""
                  type="text"
                  value=""
                />
                <span>
                  Name
                </span>
                <i />
              </div>
              <div
                class="form__inputBox"
              >
                <input
                  name="description"
                  placeholder=" "
                  required=""
                  type="text"
                  value=""
                />
                <span>
                  Description
                </span>
                <i />
              </div>
            </div>
            <div
              class="form__selectors"
            >
              <div
                class="form__selectorBox"
              >
                <span>
                  Kind
                </span>
                <select
                  name="kind"
                  placeholder=" "
                  required=""
                >
                  <option
                    value="select"
                  >
                    Select the type
                  </option>
                  <option
                    value="house"
                  >
                    House
                  </option>
                  <option
                    value="apartment"
                  >
                    Apartment
                  </option>
                </select>
              </div>
              <div
                class="form__selectorBox"
              >
                <span>
                  Guest Capacity
                </span>
                <input
                  min="1"
                  name="guest_capacity"
                  placeholder=" "
                  required=""
                  type="number"
                  value="1"
                />
              </div>
              <div
                class="form__selectorBox"
              >
                <span>
                  Bedrooms
                </span>
                <input
                  min="1"
                  name="bedrooms"
                  placeholder=" "
                  required=""
                  type="number"
                  value="1"
                />
              </div>
              <div
                class="form__selectorBox"
              >
                <span>
                  Beds
                </span>
                <input
                  min="1"
                  name="beds"
                  placeholder=" "
                  required=""
                  type="number"
                  value="1"
                />
              </div>
              <div
                class="form__selectorBox"
              >
                <span>
                  Baths
                </span>
                <input
                  min="1"
                  name="baths"
                  placeholder=" "
                  required=""
                  type="number"
                  value="1"
                />
              </div>
              <div
                class="form__selectorBox"
              >
                <span>
                  Size
                </span>
                <input
                  min="1"
                  name="size"
                  placeholder=" "
                  required=""
                  step="any"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="form__checkBox"
            >
              <h3>
                Under which categories would you match your property?
              </h3>
              <p
                class="form__checkBox__text"
              >
                (You must select at least 1 category)
              </p>
            </div>
            <div
              class="form__addressBox"
            >
              <h3>
                Enter the address of your property
              </h3>
              <div
                class="form__inputBox form__addressBox__element-full"
              >
                <input
                  name="street"
                  placeholder=" "
                  required=""
                  type="text"
                  value=""
                />
                <span>
                  Street
                </span>
                <i />
              </div>
              <div
                class="form__inputBox form__addressBox__element-half"
              >
                <input
                  min="1"
                  name="number"
                  placeholder=" "
                  required=""
                  type="number"
                  value=""
                />
                <span>
                  House Number
                </span>
                <i />
              </div>
              <div
                class="form__inputBox form__addressBox__element-half"
              >
                <input
                  name="zip_code"
                  placeholder=" "
                  required=""
                  type="text"
                  value=""
                />
                <span>
                  Zip Code
                </span>
                <i />
              </div>
              <div
                class="form__inputBox form__addressBox__element-half"
              >
                <input
                  name="city"
                  placeholder=" "
                  required=""
                  type="text"
                  value=""
                />
                <span>
                  City
                </span>
                <i />
              </div>
              <div
                class="form__inputBox form__addressBox__element-half"
              >
                <input
                  name="country"
                  placeholder=" "
                  required=""
                  type="text"
                  value=""
                />
                <span>
                  Country
                </span>
                <i />
              </div>
              <div
                class="form__inputBox form__addressBox__element-half"
              >
                <input
                  name="continent"
                  placeholder=" "
                  required=""
                  type="text"
                  value=""
                />
                <span>
                  Continent
                </span>
                <i />
              </div>
            </div>
            <button
              class="form__btn"
              type="submit"
            >
              Add your property!
            </button>
          </form>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="new-property"
    >
      <section
        class="add-property"
      >
        <div
          class="add-property__header"
        >
          <h2>
            Add your property so that other people can rent it!
          </h2>
          <p>
            Please, fill all the inputs so that you can host a home!
          </p>
        </div>
        <form
          class="form"
        >
          <div
            class="form__text-containers"
          >
            <div
              class="form__inputBox"
            >
              <input
                name="name"
                placeholder=" "
                required=""
                type="text"
                value=""
              />
              <span>
                Name
              </span>
              <i />
            </div>
            <div
              class="form__inputBox"
            >
              <input
                name="description"
                placeholder=" "
                required=""
                type="text"
                value=""
              />
              <span>
                Description
              </span>
              <i />
            </div>
          </div>
          <div
            class="form__selectors"
          >
            <div
              class="form__selectorBox"
            >
              <span>
                Kind
              </span>
              <select
                name="kind"
                placeholder=" "
                required=""
              >
                <option
                  value="select"
                >
                  Select the type
                </option>
                <option
                  value="house"
                >
                  House
                </option>
                <option
                  value="apartment"
                >
                  Apartment
                </option>
              </select>
            </div>
            <div
              class="form__selectorBox"
            >
              <span>
                Guest Capacity
              </span>
              <input
                min="1"
                name="guest_capacity"
                placeholder=" "
                required=""
                type="number"
                value="1"
              />
            </div>
            <div
              class="form__selectorBox"
            >
              <span>
                Bedrooms
              </span>
              <input
                min="1"
                name="bedrooms"
                placeholder=" "
                required=""
                type="number"
                value="1"
              />
            </div>
            <div
              class="form__selectorBox"
            >
              <span>
                Beds
              </span>
              <input
                min="1"
                name="beds"
                placeholder=" "
                required=""
                type="number"
                value="1"
              />
            </div>
            <div
              class="form__selectorBox"
            >
              <span>
                Baths
              </span>
              <input
                min="1"
                name="baths"
                placeholder=" "
                required=""
                type="number"
                value="1"
              />
            </div>
            <div
              class="form__selectorBox"
            >
              <span>
                Size
              </span>
              <input
                min="1"
                name="size"
                placeholder=" "
                required=""
                step="any"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="form__checkBox"
          >
            <h3>
              Under which categories would you match your property?
            </h3>
            <p
              class="form__checkBox__text"
            >
              (You must select at least 1 category)
            </p>
          </div>
          <div
            class="form__addressBox"
          >
            <h3>
              Enter the address of your property
            </h3>
            <div
              class="form__inputBox form__addressBox__element-full"
            >
              <input
                name="street"
                placeholder=" "
                required=""
                type="text"
                value=""
              />
              <span>
                Street
              </span>
              <i />
            </div>
            <div
              class="form__inputBox form__addressBox__element-half"
            >
              <input
                min="1"
                name="number"
                placeholder=" "
                required=""
                type="number"
                value=""
              />
              <span>
                House Number
              </span>
              <i />
            </div>
            <div
              class="form__inputBox form__addressBox__element-half"
            >
              <input
                name="zip_code"
                placeholder=" "
                required=""
                type="text"
                value=""
              />
              <span>
                Zip Code
              </span>
              <i />
            </div>
            <div
              class="form__inputBox form__addressBox__element-half"
            >
              <input
                name="city"
                placeholder=" "
                required=""
                type="text"
                value=""
              />
              <span>
                City
              </span>
              <i />
            </div>
            <div
              class="form__inputBox form__addressBox__element-half"
            >
              <input
                name="country"
                placeholder=" "
                required=""
                type="text"
                value=""
              />
              <span>
                Country
              </span>
              <i />
            </div>
            <div
              class="form__inputBox form__addressBox__element-half"
            >
              <input
                name="continent"
                placeholder=" "
                required=""
                type="text"
                value=""
              />
              <span>
                Continent
              </span>
              <i />
            </div>
          </div>
          <button
            class="form__btn"
            type="submit"
          >
            Add your property!
          </button>
        </form>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
